app/src/main/java/org/grammatek/simacorrect/spellcheckerservice/SimaCorrectSpellCheckerService.kt

```diff
@@ -100,22 +106,26 @@ class SimaCorrectSpellCheckerService : SpellCheckerService() {
 
         fun reconstructSuggestions(
             results: Array<SuggestionsInfo>,
-            resultsIndexes: Array<YfirlesturAnnotation.SuggestionIndexes>,
+            resultsIndices: Array<YfirlesturAnnotation.AnnotationIndices>,
             originalTextInfo: TextInfo
         ): SentenceSuggestionsInfo? {
-            if (results.isEmpty()) {
+            if (results.isEmpty() || originalTextInfo.text.isEmpty()) {
                 return null
             }
             val offsets = IntArray(results.size)
             val lengths = IntArray(results.size)
             val reconstructedSuggestions = arrayOfNulls<SuggestionsInfo>(results.size)
-
             for (i in results.indices) {
-                offsets[i] = resultsIndexes[i].startChar
-                lengths[i] = resultsIndexes[i].length
+                offsets[i] = resultsIndices[i].startChar
+                lengths[i] = resultsIndices[i].length
                 val result: SuggestionsInfo = results[i]
-                result.setCookieAndSequence(originalTextInfo.cookie, originalTextInfo.sequence)
-                reconstructedSuggestions[i] = result
+
+                if (result.suggestionsCount == 0) {
```

Comment:
This if else statement is here to "solve" the https://github.com/grammatek/simacorrect/issues/22 issue but as a consequence it means that we don't display annotations without suggestions.